### PR TITLE
Trap lineup

### PR DESCRIPTION
**Pull request by voidSevenSevenSix**
This is mostly taken from last year, sets the final point to drive to be lined up with the nearest apriltag

### CODE REVIEW

**Comment by voidSevenSevenSix** [comment]
Do NOT review this it is NOT done and is COMPLETELY wrong.

File: src/main/java/frc/robot/commands/TrapLineup.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Constants;
+import frc.robot.subsystems.drive.DriveSubsystem;
+import frc.robot.subsystems.intakeLimelight.IntakeLimelight;
+
+public class TrapLineup extends Command{
+  private DriveSubsystem drive;
+  private IntakeLimelight tagLimelight;
+  private boolean targetLocked = false;
+  private final double TRAP_SCORE_DISTANCE_ADJUSTMENT = Constants.PLACEHOLDER_DOUBLE;
```
Comment (balghane) [inline src/main/java/frc/robot/commands/TrapLineup.java]: Units

File: src/main/java/frc/robot/commands/TrapLineup.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import java.util.Optional;
+
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Constants;
+import frc.robot.subsystems.drive.DriveSubsystem;
+import frc.robot.subsystems.intakeLimelight.IntakeLimelight;
+
+public class TrapLineup extends Command{
```
Comment (balghane) [inline src/main/java/frc/robot/commands/TrapLineup.java]: rename? Also comment

File: src/main/java/frc/robot/commands/TrapLineup.java (hunk 1 above)
Comment (balghane) [reply src/main/java/frc/robot/commands/TrapLineup.java]: unused

File: src/main/java/frc/robot/commands/SetTrapLineupPosition.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.drive.DriveSubsystem;
+import frc.robot.subsystems.intakeLimelight.IntakeLimelight;
+
+public class SetTrapLineupPosition extends Command{
+  private DriveSubsystem drive;
+  private IntakeLimelight tagLimelight;
+  private boolean targetLocked = false;
+
+  //Set the final position in the drive subsystem to the location of the tag the limelight sees
```
Comment (balghane) [inline src/main/java/frc/robot/commands/SetTrapLineupPosition.java]: /** */ style comments for classes, variables, and functions. // style comments for commenting code. Also I think this comment is applying to the class and not the constructor

File: src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java [hunk 4]
```diff
@@ -434,13 +434,13 @@ public Pose2d extrapolatePastPoseBasedOnVelocity(double latencySec) {
...
 
-  public void setLimelightTargetFromTransform(Transform2d transform, double latencySec) {
+  public void setLimelightTargetFromTransform(Transform2d transform, double latencySec, boolean usingFrontLimelight) {
     // Transform is to get the limelight to the correct location, not to get the
     // robot
     // Here we correct for that
     Transform2d flipTransform =
         new Transform2d(
-            new Translation2d(-transform.getX(), transform.getY()), transform.getRotation());
+            new Translation2d(usingFrontLimelight?(transform.getX()):(-transform.getX()), transform.getY()), transform.getRotation()); //TODO we're gonna fix this in like 20 minutes
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java]: Is this correct as of the testing today?

Comment (balghane) [reply src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java]: Also does the TODO still apply?

**Review by balghane — APPROVED** [review]

**Pull request merged**